packages/cashc/src/ast/Globals.ts

```diff
@@ -35,47 +35,22 @@ export enum TimeOp {
 }
 
 export enum Class {
-  OUTPUT_P2SH = 'OutputP2SH',
-  OUTPUT_P2PKH = 'OutputP2PKH',
-  OUTPUT_NULLDATA = 'OutputNullData',
-}
-
-export enum PreimageField {
-  VERSION = 'tx.version', // bytes4
-  HASHPREVOUTS = 'tx.hashPrevouts', // bytes32
-  HASHSEQUENCE = 'tx.hashSequence', // bytes32
-  OUTPOINT = 'tx.outpoint', // bytes36
-  BYTECODE = 'tx.bytecode', // bytes
-  VALUE = 'tx.value', // bytes8
-  SEQUENCE = 'tx.sequence', // bytes4
-  HASHOUTPUTS = 'tx.hashOutputs', // bytes32
-  LOCKTIME = 'tx.locktime', // bytes4
-  HASHTYPE = 'tx.hashtype', // bytes4
+  LOCKING_BYTECODE_P2SH = 'LockingBytecodeP2SH',
+  LOCKING_BYTECODE_P2PKH = 'LockingBytecodeP2PKH',
+  LOCKING_BYTECODE_NULLDATA = 'LockingBytecodeNullData',
 }
 
 export const GLOBAL_SYMBOL_TABLE = new SymbolTable();
 
-// Preimage fields
-GLOBAL_SYMBOL_TABLE.set(Symbol.global(PreimageField.VERSION, new BytesType(4)));
-GLOBAL_SYMBOL_TABLE.set(Symbol.global(PreimageField.HASHPREVOUTS, new BytesType(32)));
-GLOBAL_SYMBOL_TABLE.set(Symbol.global(PreimageField.HASHSEQUENCE, new BytesType(32)));
-GLOBAL_SYMBOL_TABLE.set(Symbol.global(PreimageField.OUTPOINT, new BytesType(36)));
-GLOBAL_SYMBOL_TABLE.set(Symbol.global(PreimageField.BYTECODE, new BytesType()));
-GLOBAL_SYMBOL_TABLE.set(Symbol.global(PreimageField.VALUE, new BytesType(8)));
-GLOBAL_SYMBOL_TABLE.set(Symbol.global(PreimageField.SEQUENCE, new BytesType(4)));
-GLOBAL_SYMBOL_TABLE.set(Symbol.global(PreimageField.HASHOUTPUTS, new BytesType(32)));
-GLOBAL_SYMBOL_TABLE.set(Symbol.global(PreimageField.LOCKTIME, new BytesType(4)));
-GLOBAL_SYMBOL_TABLE.set(Symbol.global(PreimageField.HASHTYPE, new BytesType(4)));
-
 // Classes
 GLOBAL_SYMBOL_TABLE.set(
-  Symbol.class(Class.OUTPUT_P2SH, new BytesType(32), [new BytesType(8), new BytesType(20)]),
+  Symbol.class(Class.LOCKING_BYTECODE_P2SH, new BytesType(32), [new BytesType(20)]),
```

Comment:
The point of removing the `new BytesType(8)` here is that it's a separate introspection item now so might as well unify the interface around that?

Reply:
Yes exactly, the previous flow was:

Create all the output objects (amount and locking bytecode), then concat + hash those objects and compare them to `tx.hashOutputs`.

With native introspection you can access the locking bytecode and amount separately and for each separate output. So rather than comparing the *entire* outputs, we compare the locking bytecode and the amount separately. So then it makes sense to have a utility function to construct commonly used locking bytecode schemes.